### ISSUE TITLE
Validation: Add Aria-Label for Checked/Not Checked (508)

### ISSUE THREAD

**Issue by suzannemmaus** [uswds/uswds#4477]
### Describe the bug
The checkmarks that display next to each met requirement is the only communication that the user has/hasn't met the given requirements. Users who are blind or visually impaired will rely on screen readers to convey the meaning behind the checkmarks (or lack of). I would recommend adding an aria-label to each list item that communicates the status of each requirement.

Something like (using the example shown in the screenshot):
`aria-label="Use at least one uppercase character (Checked)"`
or
`aria-label="Use at least one number (Not Checked)"`

### Steps to reproduce the bug
1. Navigate to feature using Validation component
2. Listen to what is read aloud by a screenreader (e.g. on mac, VoiceOver)
  - Notice - no context is given about what is checked or not checked

### Expected behavior
Screen readers should read the content of the Validation component, communicating with the user which items have met requirements and which have not.

### Screenshots
<img width="359" alt="Screen Shot 2022-01-24 at 4 46 54 PM" src="https://user-images.githubusercontent.com/10403779/150875350-5fc5fa75-8e7d-46e2-9a78-a992d3e97ec7.png">
<img width="378" alt="Screen Shot 2022-01-24 at 4 47 58 PM" src="https://user-images.githubusercontent.com/10403779/150875367-2dbed159-1649-462c-a1f4-9b91484e9af8.png">

**Comment by fafnirical** [comment]
To add on to this, it doesn't appear like the `aria-checked` attribute that's there on the list item is working with screen readers (at least with NVDA and Windows Narrator, both with Firefox).

`axe-core` also reports [an error](https://dequeuniversity.com/rules/axe/4.3/aria-allowed-attr) on the list item when it has that attribute. To reproduce, visit https://designsystem.digital.gov/components/validation/, type anything into the checkbox (to get the attribute to be added; it isn't there on first load), and run axe (or Accessibility Insights, or whichever accessibility tool you'd like).

**Comment by aduth** [comment]
Could the [Step Indicator component](https://designsystem.digital.gov/components/step-indicator/) serve as precedent here?

In those code snippets, it's recommended to include screen reader text to indicate completion:

```html
    <li
      class="usa-step-indicator__segment usa-step-indicator__segment--complete"
    >
      <span class="usa-step-indicator__segment-label"
        >Household status <span class="usa-sr-only">completed</span></span
      >
    </li>
```

I could imagine similar could apply for the Validation component.

**Comment by tgifford-webfirst** [comment]
I got this "working" with the following setup (not tested anywhere else):
* Linux / Gnome / Orca
* Firefox
* Adding role=checkbox and tabindex=0
```html
<li class="usa-checklist__item usa-checklist__item--checked" data-validator="uppercase" aria-checked="true" 
  role="checkbox" tabindex="0">Use at least one uppercase character</li>
```
The role attribute is necessary, otherwise Firefox applies the listitem role.

I think tabindex might be appropriate because Firefox lists this error in the Accessibility info if the attribute is not present:

> Keyboard
> Error: Interactive elements must be focusable. [Learn more](https://developer.mozilla.org/docs/Web/Accessibility/Understanding_WCAG/Keyboard?utm_source=devtools&utm_medium=a11y-panel-checks-keyboard#Interactive_elements_must_be_focusable)

There's also this warning, which is odd given that the element does contain text that is read. You could probably add an aria-label attribute, but perhaps this should be viewed as a false positive, or maybe it will behave differently in other setups.
> Text Labels and Names
> Warning: Form elements should have a visible text label. [Learn more](https://developer.mozilla.org/docs/Web/Accessibility/Understanding_WCAG/Text_labels_and_names?utm_source=devtools&utm_medium=a11y-panel-checks-text-label#Form_elements_should_have_a_visible_text_label)

**Comment by tgifford-webfirst** [comment]
The PR doesn't use the role="checkbox" attribute, so my notes aren't relevant. I was using that role because the code uses the aria-checked property, which I think should only be used for focusable elements based on the roles it is used in. Based on the approach in the linked PR, I would suggest not using the aria-checked attribute. However, I'm not an expert and I haven't tested on the many many other UA/AT combinations, not even the ones that are most popular.

https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-checked#associated_roles

**Comment by mejiaj** [comment]
@tgifford-webfirst ah my mistake, sorry I misread. You make a good point, I'll put a hold label so we can do additional testing.
